An attempt to emit a .Net module throws KeyNotFoundException when Nullable feature is enabled

```
        [Fact]
        public void IgnoreEffectivelyInternalStructFieldsOfReferenceTypeFromAddedModule_PlusNullable()
        {
            var source = @"
internal class C1
{
    public struct S
    {
        public string data;
    }
}
public struct Struct
{
    internal C1.S data;
}
";
            var comp1 = CreateCompilation(source, options: WithNonNullTypesTrue(TestOptions.DebugModule));
            var moduleReference = comp1.EmitToImageReference();

            var source2 =
@"class Program
{
    public static void Main()
    {
        Struct r1;
        var r2 = r1;
    }
}";
            CreateCompilation(source2, references: new MetadataReference[] { moduleReference }, options: WithNonNullTypesTrue()).VerifyDiagnostics(
                // (6,18): error CS0165: Use of unassigned local variable 'r1'
                //         var r2 = r1;
                Diagnostic(ErrorCode.ERR_UseDefViolation, "r1").WithArguments("r1").WithLocation(6, 18)
                );
        }
```

Observed:
```
Test 'Microsoft.CodeAnalysis.CSharp.UnitTests.StructTests.IgnoreEffectivelyInternalStructFieldsOfReferenceTypeFromAddedModule_PlusNullable' failed:
	System.Collections.Generic.KeyNotFoundException : The given key was not present in the dictionary.
	at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
	PEWriter\FullMetadataWriter.cs(461,0): at Microsoft.Cci.FullMetadataWriter.DefinitionIndex`1.get_Item(T item)
	PEWriter\FullMetadataWriter.cs(179,0): at Microsoft.Cci.FullMetadataWriter.GetMethodDefinitionHandle(IMethodDefinition def)
	PEWriter\MetadataWriter.cs(745,0): at Microsoft.Cci.MetadataWriter.GetCustomAttributeTypeCodedIndex(IMethodReference methodReference)
	PEWriter\MetadataWriter.cs(2149,0): at Microsoft.Cci.MetadataWriter.AddCustomAttributeToTable(EntityHandle parentHandle, ICustomAttribute customAttribute)
	PEWriter\MetadataWriter.cs(2090,0): at Microsoft.Cci.MetadataWriter.AddModuleAttributesToTable(CommonPEModuleBuilder module)
	PEWriter\MetadataWriter.cs(2007,0): at Microsoft.Cci.MetadataWriter.PopulateCustomAttributeTableRows(ImmutableArray`1 sortedGenericParameters)
	PEWriter\MetadataWriter.cs(1942,0): at Microsoft.Cci.MetadataWriter.PopulateTypeSystemTables(Int32[] methodBodyOffsets, BlobBuilder mappedFieldDataWriter, BlobBuilder resourceWriter, BlobBuilder dynamicAnalysisDataOpt, Blob& mvidFixup)
	PEWriter\MetadataWriter.cs(1839,0): at Microsoft.Cci.MetadataWriter.BuildMetadataAndIL(PdbWriter nativePdbWriterOpt, BlobBuilder ilBuilder, BlobBuilder mappedFieldDataBuilder, BlobBuilder managedResourceDataBuilder, Blob& mvidFixup, Blob& mvidStringFixup)
	PEWriter\PeWriter.cs(66,0): at Microsoft.Cci.PeWriter.WritePeToStream(EmitContext context, CommonMessageProvider messageProvider, Func`1 getPeStream, Func`1 getPortablePdbStreamOpt, PdbWriter nativePdbWriterOpt, String pdbPathOpt, Boolean metadataOnly, Boolean isDeterministic, Boolean emitTestCoverageData, Nullable`1 privateKeyOpt, CancellationToken cancellationToken)
	Compilation\Compilation.cs(2842,0): at Microsoft.CodeAnalysis.Compilation.SerializePeToStream(CommonPEModuleBuilder moduleBeingBuilt, DiagnosticBag metadataDiagnostics, CommonMessageProvider messageProvider, Func`1 getPeStream, Func`1 getMetadataPeStreamOpt, Func`1 getPortablePdbStreamOpt, PdbWriter nativePdbWriterOpt, String pdbPathOpt, Boolean metadataOnly, Boolean includePrivateMembers, Boolean isDeterministic, Boolean emitTestCoverageData, Nullable`1 privateKeyOpt, CancellationToken cancellationToken)
	Compilation\Compilation.cs(2671,0): at Microsoft.CodeAnalysis.Compilation.SerializeToPeStream(CommonPEModuleBuilder moduleBeingBuilt, EmitStreamProvider peStreamProvider, EmitStreamProvider metadataPEStreamProvider, EmitStreamProvider pdbStreamProvider, Func`2 testSymWriterFactory, DiagnosticBag diagnostics, Boolean metadataOnly, Boolean includePrivateMembers, Boolean emitTestCoverageData, String pePdbFilePath, Nullable`1 privateKeyOpt, CancellationToken cancellationToken)
	Compilation\Compilation.cs(2432,0): at Microsoft.CodeAnalysis.Compilation.Emit(Stream peStream, Stream metadataPEStream, Stream pdbStream, Stream xmlDocumentationStream, Stream win32Resources, IEnumerable`1 manifestResources, EmitOptions options, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts, CompilationTestData testData, CancellationToken cancellationToken)
	Compilation\CompilationExtensions.cs(52,0): at Microsoft.CodeAnalysis.Test.Utilities.CompilationExtensions.EmitToArray(Compilation compilation, EmitOptions options, CompilationTestData testData, DiagnosticDescription[] expectedWarnings, Stream pdbStream, IMethodSymbol debugEntryPoint, Stream sourceLinkStream, IEnumerable`1 embeddedTexts)
	Compilation\CompilationExtensions.cs(98,0): at Microsoft.CodeAnalysis.Test.Utilities.CompilationExtensions.EmitToImageReference(Compilation comp, EmitOptions options, Boolean embedInteropTypes, ImmutableArray`1 aliases, DiagnosticDescription[] expectedWarnings)
	FlowAnalysis\StructTests.cs(427,0): at Microsoft.CodeAnalysis.CSharp.UnitTests.StructTests.IgnoreEffectivelyInternalStructFieldsOfReferenceTypeFromAddedModule_PlusNullable()
```

--- a/src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs
+++ b/src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs
@@ -537,6 +537,11 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 
             public void AddInjectedSymbols(NamespaceSymbol containingNamespace)
             {
+                if (containingNamespace.DeclaringCompilation.Options.OutputKind.IsNetModule())
+                {
+                    return;
+                }
+
                 const string codeAnalysis = "CodeAnalysis";
                 const string system = "System";
                 const string microsoft = "Microsoft";

--- a/src/Compilers/CSharp/Test/Semantic/FlowAnalysis/StructTests.cs
+++ b/src/Compilers/CSharp/Test/Semantic/FlowAnalysis/StructTests.cs
@@ -407,6 +407,42 @@ public struct Struct
                 );
         }
 
+        [Fact]
+        [WorkItem(30756, "https://github.com/dotnet/roslyn/issues/30756")]
+        public void IgnoreEffectivelyInternalStructFieldsOfReferenceTypeFromAddedModule_PlusNullable()
+        {
+            var source = @"
+internal class C1
+{
+    public struct S
+    {
+        public string data;
+    }
+}
+public struct Struct
+{
+    internal C1.S data;
+}
+";
+            var comp1 = CreateCompilation(source, options: WithNonNullTypesTrue(TestOptions.DebugModule));
+            var moduleReference = comp1.EmitToImageReference();
+
+            var source2 =
+@"class Program
+{
+    public static void Main()
+    {
+        Struct r1;
+        var r2 = r1;
+    }
+}";
+            CreateCompilation(source2, references: new MetadataReference[] { moduleReference }, options: WithNonNullTypesTrue()).VerifyDiagnostics(
+                // (6,18): error CS0165: Use of unassigned local variable 'r1'
+                //         var r2 = r1;
+                Diagnostic(ErrorCode.ERR_UseDefViolation, "r1").WithArguments("r1").WithLocation(6, 18)
+                );
+        }
+
         [Fact, WorkItem(1072447, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/1072447")]
         public void IgnorePrivateStructFieldsOfReferenceTypeFromAddedModule02()
         {

